fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/utils/PojoToRowDataConverter.java

```diff
@@ -0,0 +1,189 @@
+/*
+ *  Copyright (c) 2025 Alibaba Group Holding Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.alibaba.fluss.flink.utils;
+
+import com.alibaba.fluss.types.DataType;
+import com.alibaba.fluss.types.RowType;
+
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.TimestampData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for converting Java POJOs to Flink's {@link RowData} format.
+ *
+ * <p>This utility uses reflection to map fields from POJOs to RowData based on a given schema. It
+ * can be used by custom serialization schemas to simplify the implementation of the serialization
+ * logic.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * // Create a converter
+ * PojoToRowDataConverter<Order> converter =
+ *     new PojoToRowDataConverter<>(Order.class, rowType);
+ *
+ * // Convert a POJO to RowData
+ * Order order = new Order(1001L, 5001L, 10, "123 Mumbai");
+ * RowData rowData = converter.convert(order);
+ * }</pre>
+ *
+ * @param <T> The POJO type to convert
+ */
+public class PojoToRowDataConverter<T> {
+    private static final Logger LOG = LoggerFactory.getLogger(PojoToRowDataConverter.class);
+
+    private final Class<T> pojoClass;
+    private final RowType rowType;
+    private final Map<String, Field> fieldMap = new HashMap<>();
+
+    /**
+     * Creates a new converter for the specified POJO class and row type. * * @param pojoClass The
+     * class of POJOs to convert
+     *
+     * @param rowType The row schema to use for conversion
+     */
+    public PojoToRowDataConverter(Class<T> pojoClass, RowType rowType) {
+        this.pojoClass = pojoClass;
+        this.rowType = rowType;
+        initializeFieldMap();
+    }
+
+    private void initializeFieldMap() {
```

Comment:
Instead of analyze the pojo class using reflection by ourselves, I would suggest directly using Flink `Types.POJO(class)` to get the `PojoTypeInfo` and we can get the Java `Field` from it. If you look into the source code of `TypeExtractor.createTypeInfo`, you will find devil is in the details... 

Besides, we can't flatten all the nested fields, because there may be different fields with the same field name. Such nested POJO fields should be converted into nseted `InternalRow`. We don't need to support this case and can just throw nested POJO fields, as we currently doesn't support nested row type.